### ISSUE TITLE
bug: returning Buffer causes internal transformResult to throw

### ISSUE THREAD

**Issue by is-it-ayush** [trpc/trpc#4544]
### Provide environment information

```
  System:
    OS: Linux 5.10 Debian GNU/Linux 11 (bullseye) 11 (bullseye)
    CPU: (4) x64 Intel(R) Core(TM) i3-4130 CPU @ 3.40GHz
    Memory: 1.72 GB / 5.28 GB
    Container: Yes
    Shell: 5.8 - /usr/bin/zsh
  Binaries:
    Node: 18.16.0 - ~/.nvm/versions/node/v18.16.0/bin/node
    Yarn: 1.22.19 - ~/.nvm/versions/node/v18.16.0/bin/yarn
    npm: 9.5.0 - ~/.nvm/versions/node/v18.16.0/bin/npm
  npmPackages:
    @tanstack/react-query: ^4.29.7 => 4.29.7
    @trpc/client: ^10.26.0 => 10.26.0
    @trpc/next: ^10.26.0 => 10.26.0
    @trpc/react-query: ^10.26.0 => 10.26.0
    @trpc/server: ^10.26.0 => 10.26.0
    next: ^13.4.2 => 13.4.2
    react: 18.2.0 => 18.2.0
    typescript: ^5.0.4 => 5.0.4
```

### Describe the bug

Returning an object with a property of `Buffer` type or just `Buffer` in any kind from a trpc procedure be it mutation or query will result in the following error, `TRPCClientError: Unable to transform response from server`. It points to an error thrown within an internal `transformResult` function. The request has a valid response which is `200 OK` ie. it's not the fault of the api/procedure returning bad data/json and I've verified it as the data is being created in the database as expected. It's just the server response part that isn't working as expected.

### Link to reproduction

https://github.com/is-it-ayush/trpc_unable_to_transform_server_res

### To reproduce

1. Clone the repository.
2. Run `pnpm install`, `npx prisma db push`, `npx prisma generate`, `pnpm dev` in order.
3. Open console and click on the `Append` button to make a mutation in `Example` model.

### Additional information

For fun, I added a `console.log()` before the `response` argument within `transformResult` function before `response` was passed to `transformResultInner(response, runtime)` inside try/catch. I've noticed that the response has a `meta` property for procedures that return a Buffer type which isn't present in normal data other than Buffer. I'm not sure if this is what is causing the server response parsing to fail although I'm not experienced with trpc codebase so I'll let you be the judge.
![image](https://github.com/trpc/trpc/assets/36449128/fe3cef15-1430-4df5-9230-9f5b1515b85a)

### 👨‍👧‍👦 Contributing

- [X] 🙋‍♂️ Yes, I'd be down to file a PR fixing this bug!

**Comment by is-it-ayush** [comment]
### Update
This is an intresting edge case I encountered. It seems like this isn't an issue with tRPC itself. It's a superjson issue. Upon further investigation. I found this [superjson/issues/181](https://github.com/blitz-js/superjson/issues/181) which seems to explain this well.

tldr; superjson is hesistant to add Buffer to [supported array types](https://github.com/blitz-js/superjson/pull/155/files#diff-df7446068188621a3fa66e02d4e5c9081902f1bb2cd9092e06c59271978257d3R200) during deserialization as it's not **supported by browsers**.

### Workarounds
- Convert the Buffer to string before returning it from server.
- Extend superjson with custom registers for handling the Buffer edge case as described [here](https://github.com/blitz-js/superjson/issues/181#issuecomment-1207498257).

I'm marking this issue as closed as this isn't really a bug.
